pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -255,6 +275,21 @@ private void processNewSegment(SegmentMetadata segmentMetadata, URI finalSegment
       segmentZKMetadata.setCustomMap(segmentZKMetadataCustomMapModifier.modifyMap(segmentZKMetadata.getCustomMap()));
       if (!_pinotHelixResourceManager
           .updateZkMetadata(tableNameWithType, segmentZKMetadata, segmentMetadataZnRecord.getVersion())) {
+        _pinotHelixResourceManager.deleteSegment(tableNameWithType, segmentName);
+        throw new RuntimeException(
+            "Failed to update ZK metadata for segment: " + segmentName + " of table: " + tableNameWithType);
+      }
+    }
+
+    if (enableParallelPushProtection) {
+      // Release lock.
+      ZNRecord segmentMetadataZnRecord =
```

Comment:
It will be the same ZnRecord because we have acquired a lock, right?
However, I understand that we could do away with a redundant lookup for ZNRecord. So removed it.